### PR TITLE
MERGE AFTER 34 RELEASE - enables publish for desktop - flow skipped WIP

### PR DESCRIPTION
**Pull request by jessopb**
doPublish selects redux state for publish 
re: comments here https://github.com/lbryio/lbry-redux/pull/158

@akinwale How bad are the compatibility issues?

### CODE REVIEW

File: src/redux/actions/publish.js [hunk 1]
```diff
@@ -314,7 +352,7 @@ export const doPublish = (params: PublishParams) => (dispatch: Dispatch, getStat
   // Only pass file on new uploads, not metadata only edits.
   // The sdk will figure it out
   if (filePath) publishPayload.file_path = filePath;
-
+  console.log('pp', publishPayload)
```
Comment (akinwale) [inline src/redux/actions/publish.js]: Probably want to remove this before merging.

File: src/redux/selectors/publish.js [hunk 2]
```diff
@@ -54,8 +60,8 @@ export const selectMyClaimForUri = createSelector(
     return isStillEditing
       ? claimsById[editClaimId]
       : myClaims.find(claim =>
-          !contentName ? claim.name === claimName : claim.name === contentName || claim.name === claimName
-        );
+        !contentName ? claim.name === claimName : claim.name === contentName || claim.name === claimName
+      );
   }
 );
```
Comment (akinwale) [inline src/redux/selectors/publish.js]: What about the rest of the selectors, like `selectTakeOverAmount`? That may still be useful.

File: src/redux/actions/publish.js [hunk 3]
```diff
@@ -13,6 +12,15 @@ import {
   selectPendingById,
   selectMyClaimsWithoutChannels,
 } from 'redux/selectors/claims';
+import {
+  selectPublishFormValues,
+  selectMyClaimForUri,
+} from 'redux/selectors/publish';
+// @if TARGET='app'
```
Comment (seanyesmunt) [inline src/redux/actions/publish.js]: lbry-redux doesn't know about these comments, so they won't do anything.

File: src/redux/actions/publish.js [hunk 4]
```diff
@@ -238,47 +253,69 @@ export const doPublish = (params: PublishParams) => (dispatch: Dispatch, getStat
...
     thumbnail,
     channel,
     title,
     contentIsFree,
     fee,
     uri,
     nsfw,
-    claim,
-  } = params;
+    // claim,
```
Comment (seanyesmunt) [inline src/redux/actions/publish.js]: needed?

File: src/redux/actions/publish.js [hunk 5]
```diff
@@ -367,21 +368,22 @@ export const doCheckPendingPublishes = () => (dispatch: Dispatch, getState: GetS
 
   const checkFileList = () => {
     Lbry.claim_list().then(claims => {
+      // $FlowFixMe
       claims.forEach(claim => {
         // If it's confirmed, check if it was pending previously
         if (claim.confirmations > 0 && pendingById[claim.claim_id]) {
           delete pendingById[claim.claim_id];
-
+          // TODO fix notifications - pass as param as well?
```
Comment (seanyesmunt) [inline src/redux/actions/publish.js]: `doCheckPendingPublishes` should have success callbacks too. That's where the desktop can trigger the notification

File: src/redux/actions/publish.js [hunk 6]
```diff
@@ -13,6 +12,15 @@ import {
   selectPendingById,
   selectMyClaimsWithoutChannels,
 } from 'redux/selectors/claims';
+import {
+  selectPublishFormValues,
+  selectMyClaimForUri,
+} from 'redux/selectors/publish';
+// @if TARGET='app'
+import fs from 'fs';
+import path from 'path';
```
Comment (seanyesmunt) [inline src/redux/actions/publish.js]: After looking more I think these can be removed anyway

File: flow-typed/Publish.js [hunk 7]
```diff
@@ -17,15 +17,15 @@ declare type UpdatePublishFormData = {
...
+  bid?: string,
   bidError?: string,
   otherLicenseDescription?: string,
   licenseUrl?: string,
   licenseType?: string,
   uri?: string,
   nsfw: boolean,
 };
-
+// This was only used when we were passing params instead of selecting them
```
Comment (seanyesmunt) [inline flow-typed/Publish.js]: We can just delete this

File: src/redux/actions/publish.js (hunk 6 above)
Comment (jessopb) [reply src/redux/actions/publish.js]: @akinwale @seanyesmunt I thought @akinwale added fsAdapter for android, but desktop will still use 'fs' and 'path'. I pulled this logic into the selectAsset component for channel Edits, but publish still uses it as is?

File: src/redux/actions/publish.js (hunk 4 above)
Comment (jessopb) [reply src/redux/actions/publish.js]: nope

File: src/redux/actions/publish.js (hunk 6 above)
Comment (seanyesmunt) [reply src/redux/actions/publish.js]: I think `fs` can stay

**Comment by jessopb** [comment]
Also fixes #2589

**Pull request merged**